src/index.js

```diff
@@ -144,8 +140,20 @@ class Dotenv {
       return obj
     }, {})
 
-    // fix in case of missing
-    formatted['process.env'] = '{}'
+    // We have to stub any remaining `process.env`s due to Webpack 5 not polyfilling it anymore
+    // https://github.com/mrsteele/dotenv-webpack/issues/240#issuecomment-710231534
+    // However, if someone targets Node or Electron `process.env` still exists, and should therefore be kept
+    // https://webpack.js.org/configuration/target
+    if (
+      this.config.ignoreStub !== true &&
+      (
+        this.config.ignoreStub === false ||
+        (!target.startsWith('node') && !isMainThreadElectron(target))
+      )
```

Comment:
This logic became pretty ugly now, but it is explicitly tested so it should™️ be fine.

TL;DR: Use config if it's passed, otherwise use the target logic

Reply:
Codeclimate is complaining about it being too complex due to many lines, but it has many lines to make it more readable and less complex 😖